src/utilities/filter.js

```diff
@@ -0,0 +1,76 @@
+const sortCategory = (state, categories) => {
+    const { Apple, Samsung, Redmi, Realme, Oneplus,Oppo } = categories;
+    const filterdCategory = [];
+
+    if (
+        Apple === false &&
+        Samsung === false &&
+        Redmi === false &&
+        Realme === false &&
+        Oneplus === false &&
+        Oppo === false
+    ) {
+        return state;
+    }
+
+    if (Apple) {
+        const selectCategory = state.filter(
+            (item) => item.categoryName === "Apple"
+        );
+        filterdCategory.push(...selectCategory);
```

Comment:
You can create a function with below code and call it with the category name as argument to it. 
Makes it a DRY code.
```
const selectedCategoryFilter =(value) =>{
const arr = state.filter(
            (item) => item.categoryName === value
        );
        filterdCategory.push(...arr);
}
```

You can also go much further with this approach where you can reduce all this to 3-4 lines of code. Does the same job.